libraries/GCS_MAVLink/GCS_Common.cpp

```diff
@@ -2127,6 +2128,57 @@ void GCS_MAVLINK::send_raw_imu()
 #endif
 }
 
+void GCS_MAVLINK::send_highres_imu()
+{
+    Vector3f accel;
+    Vector3f gyro;
+#if AP_INERTIALSENSOR_ENABLED
+    const AP_InertialSensor &ins = AP::ins();
+    accel = ins.get_accel(0);
+    gyro = ins.get_gyro(0);
+#endif
+    
+    Vector3f mag;
+#if AP_COMPASS_ENABLED
+    const Compass &compass = AP::compass();
+    if (compass.get_count() >= 1) {
+        mag = compass.get_field(0) * 1000.0f; // convert to gauss
```

Comment:
Again - don't you really want data from the magnetometer being used by the current estimator?